packages/editor/src/browser/editor-manager.ts

```diff
@@ -90,17 +94,21 @@ export class EditorManagerImpl implements EditorManager, WidgetFactory {
         return this.activeObserver.onEditorChanged();
     }
 
-    canHandle(uri: URI, input?: EditorInput): number {
-        return 100;
+    canHandle(uri: URI, options?: EditorOpenerOptions): number {
+        return uri.scheme === 'file' ? 100 : 0;
     }
 
-    open(uri: URI, input?: EditorInput): Promise<EditorWidget> {
-        return this.widgetManager.getOrCreateWidget<EditorWidget>(this.id, uri.toString()).then(editor => {
+    open(uri: URI, options?: EditorOpenerOptions): Promise<EditorWidget> {
+        const trimmedUri = uri.withoutFragment().withoutQuery();
+        return this.widgetManager.getOrCreateWidget<EditorWidget>(this.id, trimmedUri.toString()).then(editor => {
             if (!editor.isAttached) {
-                this.app.shell.addWidget(editor, { area: 'main' });
+                const widgetOptions: ApplicationShell.WidgetOptions = (options && options.widgetOptions)
+                    ? options.widgetOptions
+                    : { area: 'main' };
```

Comment:
Not important, but `options && options.widgetOptions || { area: 'main' }` should work, too.